src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -228,6 +249,53 @@ Task DisposeCoreInternalAsync()
         }
     }
 
+    internal async Task<bool> RebuildProjectAsync(
+        CancellationToken cancellationToken)
+    {
+        Assumes.NotNull(_project.Services.HostObject);
+        await _projectThreadingService.SwitchToUIThread();
+
+        if (_vsSolutionBuildManager2 is null)
+        {
+            _vsSolutionBuildManager2 = await _vsSolutionBuildManagerService.GetValueAsync(cancellationToken);
+        }
```

Comment:
nit: the IDE will have a diagnostic to use this shorter "null-coalescing" form:

```suggestion
        _vsSolutionBuildManager2 ??= await _vsSolutionBuildManagerService.GetValueAsync(cancellationToken);
```